
# Step by step tutorial

In this tutorial we will explain how to run the pipeline and perform peakC analysis on the Sox2 4C-seq data from Geeven et al. (SRA files GSM2824300, GSM2824301 and GSM2824302). 


## Download the pipeline including the example files

    $ wget https://github.com/deLaatLab/pipe4C/archive/master.zip
    $ unzip master.zip
    $ cd ./pipe4C-master

> **note:** the pipe4C.R and functions.R files need to be placed in the same folder.
> The FASTQ files and viewpoint file can be found in the example folder.


## Modify the configuration file

*	Change the location in which the fragmented genome will be generated (fragFolder).
*	Change the location in which the bowtie2 index is stored.

## Run the pipeline

    Rscript pipe4C.R --vpFile=./example/VPinfo.txt --fqFolder=./example/ --outFolder=./outF/ --cores 8 --plot --wig

This will run the pipeline using 8 cores and generates a viewpoint plot and wig file as output, next to the default outputs.

## Check the report file

the report file indicates that the quality of the experiment is good.

* ~80% of the reads map to the viewpoint chromosome (fragMappedCisPercCorr).
* ~75% of the reads mapping to the viewpoint chromosome maps within 1MB from the viewpoint (cov1Mb).
* Furthermore ~90% of the mappable DpnII fragment ends within 100kb from the viewpoint (capt100Kb) have at least one read.

## Have a look at the viewpoint plots

A coverage plot of the viewpoint region is generated based on the normalized and smoothened data. The genomic region that is visualized, the height of the Y-axis, the unit of the X-axis (Mb, kb, bp) and the file type (PDF or PNG) are defined and can be adjusted in the configuration file.  In addition the quality metrics are displayed in this figure for a quick impression of the data.





